sycl/source/detail/kernel_program_cache.hpp

```diff
@@ -58,8 +58,15 @@ class KernelProgramCache {
   using ContextPtr = context_impl *;
 
   using PiKernelT = std::remove_pointer<RT::PiKernel>::type;
+
+  struct BuildResultKernel : public BuildResult<PiKernelT> {
+    std::mutex MKernelMutex;
```

Comment:
As far as I remember @sergey-semenov did something similar, see intel/llvm#1204. I bet he introduced per-program mutex which is locked when program builds. @sergey-semenov , will your change be unneeded if this one is commited?

Reply:
I don't think so. My change dealt specifically with attempting to call `sycl::program` API in parallel, which doesn't seem related to this change.

Reply:
I agree with @sergey-semenov.